packages/react-server-webpack-plugin/src/react-server-webpack-plugin.ts

```diff
@@ -81,6 +81,13 @@ function serverSource(options: Options) {
     import React from 'react';
     import {createServer} from '@shopify/react-server';
     import App from 'index';
+    process.on('uncaughtException', logError);
+    process.on('unhandledRejection', logError);
+    function logError(error) {
+      const errorLog = \`\${error.stack || error.message || 'No stack trace was present'}\`;
+      console.log(\`React Server failed to start.\n\${errorLog}\`);
```

Comment:
How confident are you about this message only being thrown during startup?  Does koa catch 100% of rejected promises?

Reply:
I'm pretty confident about this change. I tried creating some e2e tests which would involve testing dev startup which gets too messy with our e2e testing approach. Unless there's an intuitive way for us to test dev startups that I'm not aware of. 


> Does koa catch 100% of rejected promises

No, not natively to koa. However, our [logger](https://github.com/Shopify/quilt/blob/master/packages/react-server/src/logger/logger.ts#L88-L91) will handles rejected promises within koa.